File: server/routes/lobby.js
const express = require('express');
const router = express.Router();
const db = require('../db/connection');

router.post('/create', async (req, res) => {
    let id;
    let isUnique = false;

    while (!isUnique) {
        id = crypto.randomUUID();
        const [rows] = await db.execute('SELECT id FROM lobbies WHERE id = ?', [id]);
        if (rows.length === 0) {
            isUnique = true;
        }
    }

    await db.execute('INSERT INTO lobbies (id, isOpen) VALUES (?, 1)', [id]);
    res.json({ lobbyId: id });
});

router.get('/getLobby', async(req, res) =>{
    try {
        const [rows] = await db.execute('SELECT * FROM `lobbies` WHERE `isOpen` = 1');

        if(rows.length === 0 ){
            return res.status(404).json({error: "Brak lobby"});
        }

        res.status(200).json({rows});
    } catch (err) {
        console.error("Błąd serwera: ", err);
        res.status(500).json({ error: "Błąd serwera"});
    }
});
module.exports = router;

File: API/routes/lobby.py
import uuid
from flask import Blueprint, jsonify
from db import get_db_connection

lobby_bp = Blueprint('lobby', __name__)

@lobby_bp.route('/create', methods=['POST'])
def create_lobby():
    lobby_id = str(uuid.uuid4()) # Odpowiednik crypto.randomUUID()
    conn = get_db_connection()
    conn.execute('INSERT INTO lobbies (id, isOpen) VALUES (?, 1)', (lobby_id,))
    conn.commit()
    conn.close()
    return jsonify({"lobbyId": lobby_id})

@lobby_bp.route('/getLobby', methods=['GET'])
def get_lobbies():
    try:
        conn = get_db_connection()
        rows = conn.execute('SELECT * FROM lobbies WHERE isOpen = 1').fetchall()
        conn.close()
        
        if not rows:
            return jsonify({"error": "Brak lobby"}), 404
        
        return jsonify({"rows": [dict(row) for row in rows]})
    except Exception as e:
        print(f"Błąd serwera: {e}")
        return jsonify({"error": "Błąd serwera"}), 500